sql/core/src/test/scala/org/apache/spark/sql/test/DataFrameReaderWriterSuite.scala

```diff
@@ -625,15 +626,21 @@ class DataFrameReaderWriterSuite extends QueryTest with SharedSparkSession with
     spark.createDataset(data).write.mode(SaveMode.Overwrite).text(dir)
 
     // Reader, without user specified schema
-    testRead(spark.read.textFile().toDF(), Seq.empty, textSchema)
+    withTempDir { tmpDir =>
```

Comment:
@HyukjinKwon 
This is a test case for `Empty Dataset`. When reading an empty directory, you will get a `Empty Dataset`.